lxd/db/cluster/devices.mapper.go

```diff
@@ -39,12 +39,36 @@ func GetDevices(ctx context.Context, tx *sql.Tx, parent string) (map[int][]Devic
 		fillParent[i] = strings.Replace(parent, "_", "s_", -1) + "s"
 	}
 
-	sqlStmt, err := prepare(tx, fmt.Sprintf(deviceObjectsLocal, fillParent...))
-	if err != nil {
-		return nil, err
+	queryStr := fmt.Sprintf(deviceObjectsLocal, fillParent...)
+	queryParts := strings.SplitN(queryStr, "ORDER BY", 2)
+	args := make([]any, 0, DqliteMaxParams)
+
+	for i, filter := range filters {
+		var cond string
+		if i == 0 {
+			cond = " WHERE ( %s )"
+		} else {
+			cond = " OR ( %s )"
+		}
+
+		entries := []string{}
+		if filter.Name != nil {
+			entries = append(entries, "name = ?")
+			args = append(args, filter.Name)
+		}
+
+		if filter.Type != nil {
+			entries = append(entries, "type = ?")
+			args = append(args, filter.Type)
+		}
+
```

Comment:
This function seems to be missing a check to fail when a filter is passed in that has both Name and Type set to nil. Unlike the other functions which will fail.